pkg/config/otlp.go

```diff
@@ -13,6 +13,7 @@ const (
 	OTLPTracesEnabled         = OTLPSection + "." + OTLPTracesSubSectionKey + ".enabled"
 	OTLPReceiverSubSectionKey = "receiver"
 	OTLPReceiverSection       = OTLPSection + "." + OTLPReceiverSubSectionKey
+	OTLPCensusReceiverSection = OTLPSection + ".opencensus"
```

Comment:
Since the OTLP receiver and the OpenCensus receiver are separate components, it feels weird to call one "receiver" and the other just "opencensus". I think it would make sense to match the `receivers: [receiver_name]:` pattern, but not sure if this would be too difficult to refactor in our config file.

Another possibility would be to rename this to `OTLPSection + ".opencensus_receiver"` to be more explicit.

Reply:
I agree, but that would be a breaking change, and I was thinking that we don't want to do that now for a feature that we aren't even advertising or putting in our change log. Furthermore, we are going to create a more elaborate component out of this collector, which will receive more advanced configurations (hopefully more on par with the actual Collector). At that point, we will require a big breaking change, so I thought it'd be better to avoid breaking changes on this config twice, especially for a minute unadvertised feature like this one.

But I'm happy to hear other points of view too.

Reply:
Morally, this feels like it should not be in the `otlp_config` section at all, since it has (from the user's point of view) nothing to do with OTLP. But since this is meant to be an internal only feature and there is, at least in the current state of the PR, a dependency between the OTLP config and OpenCensus (OC won't work it OTLP is not enabled AFAICT) I think it's fine to leave it here.